### PR TITLE
Remove property validations for lambda resources

### PR DESCRIPTION
**Pull request by xazhao**
### Description of changes
A customer is reporting `AWS::Lambda::Function` supports string-type integers for memory size but `AWS::Serverless::Function` doesn't support it. So it's time to get rid of the property validations.

### Description of how you validated changes
Deploying the transformed templates.

### Checklist

- [ ] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [x] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [ ] Using correct values
    - [ ] Using wrong values
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/model/lambda_.py [hunk 1]
```diff
@@ -1,38 +1,37 @@
...
-        "Layers": PropertyType(False, list_of(one_of(IS_STR, IS_DICT))),
-        "ReservedConcurrentExecutions": PropertyType(False, any_type()),
-        "FileSystemConfigs": PropertyType(False, list_of(IS_DICT)),
-        "CodeSigningConfigArn": PropertyType(False, IS_STR),
-        "ImageConfig": PropertyType(False, IS_DICT),
-        "Architectures": PropertyType(False, list_of(one_of(IS_STR, IS_DICT))),
-        "SnapStart": PropertyType(False, IS_DICT),
-        "EphemeralStorage": PropertyType(False, IS_DICT),
-        "RuntimeManagementConfig": PropertyType(False, IS_DICT),
+        "Code": PassThroughProperty(True),
```
Comment (hoffa) [inline samtranslator/model/lambda_.py:11]: These can all just be `GeneratedProperty()`. We don't need to do any runtime validation.

**Review by GavinZZ — APPROVED** [review]

**Review by ssenchenko — APPROVED** [review]

File: samtranslator/model/lambda_.py (hunk 1 above)
Comment (aahung) [reply samtranslator/model/lambda_.py:11]: +1

Comment (xazhao) [reply samtranslator/model/lambda_.py:11]: Oh right! Yes `GeneratedProperty()` is better.

**Review by aaythapa — APPROVED** [review]

File: samtranslator/model/lambda_.py [hunk 2]
```diff
@@ -1,38 +1,37 @@
...
-        "Layers": PropertyType(False, list_of(one_of(IS_STR, IS_DICT))),
-        "ReservedConcurrentExecutions": PropertyType(False, any_type()),
-        "FileSystemConfigs": PropertyType(False, list_of(IS_DICT)),
-        "CodeSigningConfigArn": PropertyType(False, IS_STR),
-        "ImageConfig": PropertyType(False, IS_DICT),
-        "Architectures": PropertyType(False, list_of(one_of(IS_STR, IS_DICT))),
-        "SnapStart": PropertyType(False, IS_DICT),
-        "EphemeralStorage": PropertyType(False, IS_DICT),
-        "RuntimeManagementConfig": PropertyType(False, IS_DICT),
+        "Code": GeneratedProperty(True),
```
Comment (hoffa) [inline samtranslator/model/lambda_.py:11]: Just `GeneratedProperty()`, as-is.

Comment (xazhao) [reply samtranslator/model/lambda_.py]: Yep used it wrong...

**Review by hoffa — APPROVED** [review]

**Review by aahung — APPROVED** [review]

**Pull request merged**